hello-world/wrapper/typescript/shims/shims.js

```diff
@@ -0,0 +1,251 @@
+const textEncoderShims = require("./textEncoder");
+const msgpackShims = require("./msgpack");
+
+const packagesShim = `
+${textEncoderShims}
+
+${msgpackShims}
+
+function msgpackEncodeValue(value) {
+  return {
+    value: Array.from(encode(value)),
+  };
+}
+
+//HACK: remove this once codegen is fixed
+const __old_subinvoke = __wrap_subinvoke;
+__wrap_subinvoke = function (plugin, method, args) {
+  if (Array.isArray(args)) {
+    return __old_subinvoke(plugin, method, args);
+  } else {
+    return __old_subinvoke(plugin, method, clean(args));
+  }
+};
+
+function clean(obj, root = true) {
+  const x = JSON.stringify(Array.from(encode(obj)));
+  __wrap_debug_log(x);
+  return JSON.parse(x);
+}
+
+const console = {
+  log: function(str) {
+    __wrap_debug_log(str);
+  },
+};
+
+function require(lib) {
+  function wrap(objName, obj) {
+    const origin = {};
+    return new Proxy(origin, {
+      get(_, name) {
+        if (obj[name]) {
+          return obj[name];
+        } else {
+          throw new Error(\`No method \${name} in \${objName}\`);
+        }
+      },
+    });
+  }
+
+  const fs = {
```

Comment:
A bit out of scope of this PR, I just want to suggest that we eventually hide everything but the shim declarations from users. My thought is that they should only need to worry about declaring shims, and not about the other content in this file.